src/test/java/com/iexec/worker/compute/ComputeControllerTests.java

```diff
@@ -71,35 +73,77 @@ void sendExitCauseForComputeComputeStage() {
         when(workerpoolAuthorizationService.isSignedWithEnclaveChallenge(CHAIN_TASK_ID, AUTH_HEADER))
                 .thenReturn(true);
 
-        final ResponseEntity<?> response =
+        // Test single cause via deprecated endpoint
+        final ResponseEntity<?> singleResponse =
                 computeController.sendExitCauseForGivenComputeStage(AUTH_HEADER,
                         ComputeStage.PRE,
                         CHAIN_TASK_ID,
                         new ExitMessage(CAUSE));
-        Assertions.assertTrue(response.getStatusCode().is2xxSuccessful());
-        Assertions.assertEquals(HttpStatus.OK.value(), response.getStatusCode().value());
+        assertThat(singleResponse.getStatusCode().is2xxSuccessful()).isTrue();
+        assertThat(HttpStatus.OK.value()).isEqualTo(singleResponse.getStatusCode().value());
+
+        // Test single cause via new bulk endpoint
+        final ResponseEntity<Void> bulkSingleResponse = computeController.sendExitCausesForGivenComputeStage(
+                AUTH_HEADER,
+                ComputeStage.POST,
+                CHAIN_TASK_ID,
+                List.of(CAUSE)
+        );
+        assertThat(HttpStatus.OK.value()).isEqualTo(bulkSingleResponse.getStatusCode().value());
+
+        // Test multiple causes via new bulk endpoint
+        List<ReplicateStatusCause> multipleCauses = List.of(
+                ReplicateStatusCause.PRE_COMPUTE_DATASET_URL_MISSING,
+                ReplicateStatusCause.PRE_COMPUTE_INVALID_DATASET_CHECKSUM
+        );
+        final ResponseEntity<Void> bulkMultipleResponse = computeController.sendExitCausesForGivenComputeStage(
+                AUTH_HEADER,
+                ComputeStage.PRE,
+                CHAIN_TASK_ID,
+                multipleCauses
+        );
+        assertThat(HttpStatus.OK.value()).isEqualTo(bulkMultipleResponse.getStatusCode().value());
```

Comment:
You can improve with `@ParameterizedTest` here

Reply:
We want to avoid testing different methods in the same unit test, this is not a good practice